pkg/proctree/thread.go

```diff
@@ -0,0 +1,41 @@
+package proctree
+
+import "sync"
+
+// Thread represents a thread.
+type Thread struct {
```

Comment:
Yep, I started doing the Thread with all the TaskInfo fields as well, and removed. I can put the TaskInfo back to Threads (this way you can have versioned information about what happened to the thread) but check what I thought:

One reason why I considered threads important is because of events coming directly from threads. My idea of only having the parent hash was because:

- Event came from thread ? Find its parent (will always be a process)
- Put the "resource" on the parent Process structure (sockets accessed)

This way threads would only be used to find the real process and save all information under that process (so you have a centralized view of what happened from the main thread or any other).

WDYT ?

Reply:
Its fine with the current details, but what will happen in the future when we want thread-specific info?
For now though you can keep it this way, but a thread must contain the following anyway - `tid`, `nsTid` and `name`.